src/core/service.c

```diff
@@ -1426,6 +1426,26 @@ static int service_spawn(
         if (r < 0)
                 return r;
 
+        if (IN_SET(s->kill_context.kill_mode, KILL_MIXED, KILL_CONTROL_GROUP) && !s->kill_context.send_sigkill) {
+                _cleanup_free_ char *text = NULL;
+                _cleanup_free_ char *f = NULL;
+                r = cg_get_path(SYSTEMD_CGROUP_CONTROLLER, UNIT(s)->cgroup_path, "cgroup.procs", &f);
+                if (r < 0)
+                        return r;
```

Comment:
please use cg_is_empty_recursive() for this check

Reply:
I ended up getting `unit_warn_leftover_processes` to identify the process existing rather than separate call to `cg_is_empty_recursive`. This didn't affect the other calls to `unit_warn_leftover_processes`.